helix-term/src/commands.rs

```diff
@@ -1000,31 +1000,70 @@ fn goto_file_vsplit(cx: &mut Context) {
     goto_file_impl(cx, Action::VerticalSplit);
 }
 
+fn resolve_selected_path(
+    selection_text: String,
+    doc_path_parent: Option<&Path>,
+) -> Option<PathBuf> {
+    let trimmed = selection_text.trim();
+    if trimmed.is_empty() {
+        return None;
+    }
+
+    let selected_path = PathBuf::from(trimmed);
+    match (selected_path.is_relative(), doc_path_parent) {
+        (true, Some(parent_path)) => {
+            let mut result = PathBuf::new();
+            result.push(parent_path);
+            result.push(selected_path);
+            Some(result)
+        }
+        _ => Some(selected_path),
+    }
+}
+
 fn goto_file_impl(cx: &mut Context, action: Action) {
     let (view, doc) = current_ref!(cx.editor);
     let text = doc.text();
     let selections = doc.selection(view.id);
-    let mut paths: Vec<_> = selections
-        .iter()
-        .map(|r| text.slice(r.from()..r.to()).to_string())
-        .collect();
     let primary = selections.primary();
+    let doc_path_parent = doc.path().and_then(|path| path.parent());
+
     if selections.len() == 1 && primary.to() - primary.from() == 1 {
-        let current_word = movement::move_next_long_word_start(
+        let current_word = textobject::textobject_word(
             text.slice(..),
-            movement::move_prev_long_word_start(text.slice(..), primary, 1),
+            primary,
+            textobject::TextObject::Inside,
             1,
+            true,
         );
```

Comment:
Looks reasonable to me, but I am surprised it even works for CJK characters.

Reply:
At least for nix it would be cool to use language text objects to get the current path (that is a type in nix)/string content. But I wasn't able to digest the tree-sitter API yet.

Reply:
We could consider adding an extra textobject for paths. I think the word textobject is a reasonable choice for the fallback behavior here though.